spec/payers_spec.rb

```diff
@@ -0,0 +1,184 @@
+                build_activity_formula(
+                  "percentage_calculated", " (percentage/100) * 0.20",
+                  "should only export when percentage is filled"
+                ),
+                build_activity_formula(
+                  "payer_payment", "to_pay * percentage_calculated",
+                  ""
+                )
+              ]
+            )
+          ]
+        )
+      ]
+    )
+  end
+
+  let(:orgunit_id) do
+    "orgunit_id"
+  end
+
+  let(:orgunits_full) do
+    [
+      Orbf::RulesEngine::OrgUnit.with(
+        ext_id:        orgunit_id,
+        path:          "path",
+        name:          "name",
+        group_ext_ids: ["G_ID_1"]
+      )
+    ]
+  end
+
+  let(:org_unit_groups) do
+    [
+      Orbf::RulesEngine::OrgUnitGroup.with(
+        ext_id: "G_ID_1",
+        name:   "group health center",
+        code:   "health_center"
+      )
+    ]
+  end
+
+  let(:dhis2_values) do
+    [
+      {
+        "dataElement"         => "dhis2_to_pay",
+        "categoryOptionCombo" => "default",
+        "value"               => 10_000,
+        "period"              => period,
+        "orgUnit"             => orgunit_id
+      },
+      {
+        "dataElement"         => "dhis2_spread_percentage",
+        "categoryOptionCombo" => "dhis2_payer_1",
+        "value"               => 10,
+        "period"              => period,
+        "orgUnit"             => orgunit_id
+      },
+      {
+        "dataElement"         => "dhis2_spread_percentage",
+        "categoryOptionCombo" => "dhis2_payer_2",
+        "value"               => 60,
+        "period"              => period,
+        "orgUnit"             => orgunit_id
+      }, {
+        "dataElement"         => "dhis2_spread_percentage",
+        "categoryOptionCombo" => "dhis2_payer_3",
+        "value"               => 30,
+        "period"              => period,
+        "orgUnit"             => orgunit_id
+      }
+    ]
+  end
+
+  it "builds problem per category option combo" do
+    solved = build_and_solve(orgunits_full, dhis2_values)
+    problem = solved.solver.build_problem
+    puts JSON.pretty_generate(problem)
+
+    expect(problem["facility_spread_01_dhis2_payer_1_exportable_for_orgunit_id_and_2016q1"]).to eq(
+      "facility_spread_01_dhis2_payer_1_percentage_is_null_for_orgunit_id_and_2016q1 == 0"
+    )
+    expect(problem["facility_spread_01_dhis2_payer_1_percentage_calculated_for_orgunit_id_and_2016q1"]).to eq(
+      "(facility_spread_01_dhis2_payer_1_percentage_for_orgunit_id_and_2016q1/100) * 0.20"
+    )
+    expect(problem["facility_spread_01_dhis2_payer_1_payer_payment_for_orgunit_id_and_2016q1"]).to eq(
+      "facility_spread_01_to_pay_for_orgunit_id_and_2016q1 * facility_spread_01_dhis2_payer_1_percentage_calculated_for_orgunit_id_and_2016q1"
+    )
+  end
+
+  it "should solve equations" do
+    solved = build_and_solve(orgunits_full, dhis2_values)
+
+    puts JSON.pretty_generate(solved.exported_values)
```

Comment:
Can we hide this behind a verbose/debug flag? The less the tests output the more anxious I am when they do start outputting a lot.

Reply:
replace with expect ;)